src/Compilers/CSharp/Portable/BoundTree/InterpolatedStringBuilderData.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed record InterpolatedStringBuilderData(
```

Comment:
>InterpolatedStringBuilderData [](start = 27, length = 29)

What record features do we actually take advantage of here? Given how much metadata is going to be produced for this type, we better use all of it. Otherwise, I think we should find another alternative. We also should review other "records" recently added to the compiler.